ocaml/xapi/message_forwarding.ml

```diff
@@ -4329,15 +4330,23 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
       info "Cluster_host.destroy";
       let local_fn = Local.Cluster_host.destroy ~self in
       let host = Db.Cluster_host.get_host ~__context ~self in
-      do_op_on ~__context ~local_fn ~host
-        (fun session_id rpc -> Client.Cluster_host.destroy rpc session_id self)
+      let cluster = Db.Cluster_host.get_cluster ~__context ~self in
+      Xapi_cluster_helpers.with_cluster_operation ~__context ~self:cluster ~doc:"Cluster_host.destroy" ~op:`remove
+        (fun () ->
+          do_op_on ~__context ~local_fn ~host
+            (fun session_id rpc -> Client.Cluster_host.destroy rpc session_id self)
+        )
 
     let force_destroy ~__context ~self =
       info "Cluster_host.force_destroy";
       let local_fn = Local.Cluster_host.force_destroy ~self in
       let host = Db.Cluster_host.get_host ~__context ~self in
-      do_op_on ~__context ~local_fn ~host
-        (fun session_id rpc -> Client.Cluster_host.force_destroy rpc session_id self)
+      let cluster = Db.Cluster_host.get_cluster ~__context ~self in
+      Xapi_cluster_helpers.with_cluster_operation ~__context ~self:cluster ~doc:"Cluster_host.force_destroy" ~op:`destroy
```

Comment:
This should be `remove`. `destroy` on the cluster object is for destroying the whole cluster.